### ISSUE TITLE
Recording stops when opening a share

### ISSUE THREAD

**Issue by zstadler** [IsraelHikingMap/Site#816]
## What I expect to happen
Recording should continue until stopped by the user

## What really happened
Recording stops after opening a share

## What I did that caused the issue - step by step
1. Start recording
2. My shares -> show a share on the map with the "eye" button
3. The GPS position keeps updating, the record button remains active, but the recording line stops

## My environment
* Operating System and version: Android 8.0.0
* Web Browser and version: App version 7.2.29

## Thing I think the developers should know, images, links etc.
![image](https://user-images.githubusercontent.com/1304610/45588952-f4fdc380-b925-11e8-8d29-41f0b7a3ff4a.png)

This may be just one of several potential causes for the recording to stop. I was unable to identify additional reasons yet. However, I have a feeling it is related to increasing the load on the application by bringing it to the foreground, resuming GPS tracking, manual zoom/pan, etc. This may have also been worsened by the poor network condition during the trip.

**Comment by HarelM** [comment]
Good catch, When selecting a share to view it makes this share the active route and thus changes the state of the recording to be readonly - which stops listening to GPS updates.
I'm not entirely sure how to solve this as this can easily happen when adding a new route and selecting it as active.
i.e. there are a lot of places when the selected route is changed and might cause the recording to stop. I need to think if the right place to handle it is the `RoutesService.selectRoute(...)`.
Might not have been a right approach to use the exiting mechanism for recording... :-/

**Comment by HarelM** [comment]
Nope, this won't work as you won't be able to add private POIs.
Damn...
